### PR TITLE
feat: Whitelisted namespace in UI

### PR DESCRIPTION
**Pull request by rachelwang20**
Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [x] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Optional. My organization is added to USERS.md.
* [ ] I've signed the CLA and my build is green ([troubleshooting builds](https://argoproj.github.io/argo-cd/developer-guide/ci/)). 

Issue link: #2900

### CODE REVIEW

**Comment by jannfis** [comment]
Seems I missed the UI part when updating the contribution guide. Fix underway @ #3315. Sorry for that, @rachelwang20

File: ui/src/app/settings/components/project-details/project-details.tsx [hunk 1]
```diff
@@ -573,6 +574,30 @@ export class ProjectDetails extends React.Component<RouteComponentProps<{name: s
                     </div>
                 )}
 
+                <h4>Whitelisted namespaced resources {helpTip('Namespace-scoped K8s API Groups and Kinds which are permitted from being deployed')}</h4>
```
Comment (alexmt) [inline ui/src/app/settings/components/project-details/project-details.tsx]: Please update as following for consistency:

```suggestion
                <h4>Whitelisted namespaced resources {helpTip('Namespace-scoped K8s API Groups and Kinds which are permitted to deploy')}</h4>
```

File: ui/src/app/settings/components/project-edit-panel/project-edit-panel.tsx [hunk 2]
```diff
@@ -160,6 +160,37 @@ export const ProjectEditPanel = (props: {nameReadonly?: boolean; defaultParams?:
                         </a>
                     </React.Fragment>
 
+                    <React.Fragment>
+                        <h4>Whitelisted Namespaced Resources</h4>
+                        <div>
+                            Namespace-scoped K8s API Groups and Kinds which are <strong>permitted</strong> from being deployed
```
Comment (alexmt) [inline ui/src/app/settings/components/project-edit-panel/project-edit-panel.tsx]:
```suggestion
                            Namespace-scoped K8s API Groups and Kinds which are <strong>permitted</strong> to deploy
```

**Review by alexmt — CHANGES_REQUESTED** [review]
Thank you for PR @rachelwang20 . Added two minor comments. Also please resolve linter warning:

```
yarn run v1.19.1
$ tslint -p ./src/app

ERROR: /Users/amatyushentsev/root/go/src/github.com/argoproj/argo-cd/ui/src/app/settings/components/project-edit-panel/project-edit-panel.tsx:21:1 - Exceeds maximum line length of 180
ERROR: /Users/amatyushentsev/root/go/src/github.com/argoproj/argo-cd/ui/src/app/settings/components/project-edit-panel/project-edit-panel.tsx:21:29 - Replace `sourceRepos:·[],·destinations:·[],·roles:·[],·syncWindows:·[],·clusterResourceWhitelist:·[],·namespaceResourceBlacklist:·[],·namespaceResourceWhitelist:·[],·...props.defaultParams` with `⏎················sourceRepos:·[],⏎················destinations:·[],⏎················roles:·[],⏎················syncWindows:·[],⏎················clusterResourceWhitelist:·[],⏎················namespaceResourceBlacklist:·[],⏎················namespaceResourceWhitelist:·[],⏎················...props.defaultParams⏎············`
```

**Review by alexmt — APPROVED** [review]
LGTM, thank you!

**Pull request merged**